### PR TITLE
fix: Retry request when ChunkedEncodingError occurred.

### PR DESCRIPTION
**Pull request by PatrykGala**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/common/backends/utils.py [hunk 1]
```diff
@@ -100,6 +101,7 @@ def wrapper(*args, **kwargs):
                 HTTPTooManyRequests,
                 HTTPInternalServerError,
                 NewConnectionError,
+                ChunkedEncodingError,
```
Comment (aniezurawski) [inline src/neptune/common/backends/utils.py:104]: How about retrying only once like in case of SSL error?

Comment (aniezurawski) [reply src/neptune/common/backends/utils.py:104]: Also some very explanatory comment would be welcome.

Comment (PatrykGala) [reply src/neptune/common/backends/utils.py:104]: I suppose, that client closes the connection before forking and after resuming - client doesn't receive a response from server.

This error is similar to `HTTPRequestTimeout or BravadoConnectionError` - and we can retry this request.

Comment (aniezurawski) [reply src/neptune/common/backends/utils.py:104]: It's hard to believe that it throws some very random exception when the connection is just closed. Can we test this hypothesis?

**Comment by mcochner** [comment]
Do you know if there is an ETA on including this in a new release of `neptune-client`? We observe issues with the stability of `neptune-client` and (quite infrequently but usually once a day) it dies on `ChunkedEncodingError`.

**Comment by aniezurawski** [comment]
@mcochner see these response: https://github.com/neptune-ai/neptune-client/issues/1396#issuecomment-1649826972
These issues are related and this workaround should help with this error as well.

**Comment by PatrykGala** [comment]
> Do you know if there is an ETA on including this in a new release of `neptune-client`? We observe issues with the stability of `neptune-client` and (quite infrequently but usually once a day) it dies on `ChunkedEncodingError`.

Hi @mcochner, 
I have been trying to reproduce this error, but without success. Could you please check your logs for any occurrences of ChunkedEncodingError along with timestamps? I would like to verify this exception with our backend logs.

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**